test/ssl-tests/20-cert-select.conf.in

```diff
@@ -87,7 +87,7 @@ our @tests = (
             "MaxProtocol" => "TLSv1.2",
         },
         test   => {
-            "ExpectedServerCertType" =>, "RSA-PSS",
+            "ExpectedServerCertType" =>, "RSA",
```

Comment:
Hmm, but something is not right here, since this is supposed to be using the ```$server_pss``` block, which actually should be an RSA-PSS key in the cert.  Need to dig further...

Reply:
The story here is actually interesting, but in a depressing sort of way ... even when the client configuration directs it to use the rsa_pss_pss_sha256 sigalg, it actually ends up wending 0x0804 (rsa_pss_rsae_sha256) on the wire.  This is because tls1_set_sigalgs() gets an array of sigNID+hashNID pairs, and goes through the lookup table to find matching SIGALG_LOOKUP entries.  Now that the TLS spec has split the RSA ones up by key OID, there are two matching table elements, and we pick the first (wrong) one.  This sigNID+hashNID format is intrinsic to our public APIs SSL{,_CTX}_set1_sigalgs{,_list}(), so the correct behavior will need to be implemented as a new public API (and SSL_ctrl case?  See also #1319) and a decision made on which way the existing API should disambiguate.
Oh, and s_client/s_server need to learn to use the new API.